src/Simplex/Messaging/Protocol.hs

```diff
@@ -300,6 +301,9 @@ data Command (p :: Party) where
   PING :: Command Sender
   -- SMP notification subscriber commands
   NSUB :: Command Notifier
+  PROXY :: SMPServer -> Maybe BasicAuth -> Command Sender -- request a relay server connection by URI
+  PHS :: Command Sender -- identify connection as a proxy, set up shared key
+  PFWD :: C.PublicKeyX25519 -> ByteString -> Command Sender -- use CorrId as CbNonce
```

Comment:
```suggestion
  PXY :: SMPServer -> Command Sender -- request a relay server connection by URI
  PPXY :: C.PublicKeyX25519 -> Command Sender -- identify connection as a proxy, set up shared key
  PFWD :: C.PublicKeyX25519 -> ByteString -> Command Sender -- use CorrId as CbNonce
```

PPXY must have a key for the session encryption

Reply:
possibly type encoded lengths / newtypes instead of byte strings to wrap/unwrap correctly. The information about encryption layers should be encoded in types, or it's really easy to get confused here...

Reply:
or was BasicAuth here to allow usage of proxy?